### DISCUSSION TITLE
[Question] Clarifications needed regarding auto scaling based on the Redis Stream

### DISCUSSION THREAD

**Discussion by vnmnm** [kedacore/keda#4488] [Q&A / Need Help]
Currently Keda supports scaling out pods based on the pending entries length in Redis Stream per consumer group. Pending entries are those that are dequeued by the consumers but not acknowledged yet. Eventually the consumers will acknowledge these once they process the entry after which the entry will be deleted from pending list. So the pending list length only depends on how many entries the consumers are dequeuing and how long they take to acknowledge. But what if the publisher publishes at a rate higher than the consumers can process? In this case the Redis Stream Length (https://redis.io/commands/xlen/) increases and wouldn't that be an ideal metric to scale consumers out upon. However, XLEN also includes the pending messages and so the correct metric would be (XLEN - XPENDING) right?

**Comment by zroubalik** [comment]
I am not a Redis expert, but that seems like a correct assumption.

**Reply by vnmnm** [reply]
@zroubalik @abhirockzz  There's some discussion regarding this [here](https://github.com/kedacore/keda/issues/746). In the current form this scaler is not useful. Are there any plans to add additional params other than pending entries count? In our case we have only one consumer group and messages are acknowledged and deleted after they are processed and hence scaling out based on the length of the stream (XLEN) will be helpful too.

**Reply by zroubalik** [reply]
@vnmnm okay, agree. Are you willing to contribute this feature?

**Reply by vnmnm** [reply]
@zroubalik I would have loved to contribute but I am not familiar with Go programming language.

**Reply by zroubalik** [reply]
@vnmnm one can always start, it is not that hard :) Btw https://github.com/kedacore/keda/pull/4592
